### DISCUSSION TITLE
3DES encryption algorithm

### DISCUSSION THREAD

**Discussion by mavis-yang** [strongswan/strongswan#414] [General]
Hi experts,

Hope you are doing well :)

I would like to confirm 3DES encryption algorithm key length in strongswan.

About 3DES algorithm mentioned in [IKEv2CipherSuites](https://wiki.strongswan.org/projects/strongswan/wiki/IKEv2CipherSuites) is as 168 bit 3DES-EDE-CBC, while in [IpsecStandards ](https://wiki.strongswan.org/projects/strongswan/wiki/IpsecStandards) referring to [RFC2451](https://datatracker.ietf.org/doc/html/rfc2451), [RFC7296](https://datatracker.ietf.org/doc/html/rfc7296), 3DES is with 192-bit key size.
From my understanding, 168 bit 3DES-EDE-CBC and 3DES-CBC are different for strongswan, right?

If 168 bit 3DES-EDE-CBC is strongswan's implementation, do you mind pointing out which standard is referred?

PS. please help remove the page in strongswan wiki https://wiki.strongswan.org/projects/strongswan/wiki/3DES_encryption_algorithm/1, I added by mistake. Sorry for your inconvenience.

Regards,
Yang Li

**Comment by tobiasbrunner** [comment]
>  From my understanding, 168 bit 3DES-EDE-CBC and 3DES-CBC are different for strongswan, right?

No. While 192 bits of key material is generated for 3DES, only 168 bits are actually used. 8 bits per key are considered parity bits, so each is only 56 bits. This is actually mentioned in [section 2.2. of RFC 2451](https://datatracker.ietf.org/doc/html/rfc2451#section-2.2).

Anyway, you really should not be using 3DES anymore.

**Reply by mavis-yang** [reply]
Tobias,
Thanks for your info.
I did not get the 168 bit is composed like this.

This topic is OK.

Regards,
Yang Li
